Plugins/BranchTree/BranchTreeForm.cs

```diff
@@ -0,0 +1,137 @@
+﻿using GitUIPluginInterfaces;
+using ResourceManager;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Windows.Forms;
+
+namespace BranchTree
+{
+    public partial class BranchTreeForm : GitExtensionsFormBase
+    {
+        TreeNode _tagTreeNode, _localTreeNode, _remoteTreeNode;
+        private readonly IGitUICommands _gitUiCommands;
+               
+
+        public BranchTreeForm()
+        {
+            InitializeComponent();
+            Translate();
+        }
+
+        public BranchTreeForm(GitUIBaseEventArgs gitUiCommands) : this()
+        {
+            _gitUiCommands = gitUiCommands.GitUICommands;
+            var gitModule = gitUiCommands != null ? gitUiCommands.GitModule : null;
+            
+            BranchesTreeView.BeginUpdate();
+            BranchesTreeView.Nodes.Clear();
+
+            _localTreeNode = BranchesTreeView.Nodes.Add("Local");
+            _remoteTreeNode = BranchesTreeView.Nodes.Add("Remote");
+            _tagTreeNode = BranchesTreeView.Nodes.Add("Tag");
+
+            InitLocalBranches(_localTreeNode, gitModule);
+            InitRemoteBranches(_remoteTreeNode, gitModule);
+            InitTag(_tagTreeNode, gitModule);
+          
+            BranchesTreeView.EndUpdate();
+            BranchesTreeView.NodeMouseDoubleClick += Branches_NodeMouseDoubleClick;
```

Comment:
You don't want to do any heavy lifting in the constructor, otherwise it leads to unresponsive UI. Constructors are there to initialise the instance and set its state, not to perform actions.
Typically you do this in `ApplicationIdle` or form's `OnLoad` event.
I would also suggest turning `InitXxx` methods into async.
And lastly, it is a very good idea to wrap paired operations in `try...finally` construct to ensure both commands execute, e.g.:

```
try 
{
    treview.BeginUpdate();
    // ...
}
finally
{
    treeview.EndUpdate();
} 
```